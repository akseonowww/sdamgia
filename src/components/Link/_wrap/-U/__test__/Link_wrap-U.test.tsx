import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import LinkWrapU from '..'

describe('Link_wrap-U', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(<LinkWrapU>Link with more clickable zone</LinkWrapU>, container)
    })
    expect(container && container.textContent).toBe(
      'Link with more clickable zone'
    )

    act(() => {
      render(
        <LinkWrapU className="Link_pseudo-U Link_pseudoBlack-U">
          Link with more clickable zone
        </LinkWrapU>,
        container
      )
    })
    expect(container && container.textContent).toBe(
      'Link with more clickable zone'
    )
  })

  it('should render a content', () => {
    act(() => {
      render(<LinkWrapU>Link with more clickable zone</LinkWrapU>, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <LinkWrapU className="Link_pseudo-U Link_pseudoBlack-U">
          Link with more clickable zone
        </LinkWrapU>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
