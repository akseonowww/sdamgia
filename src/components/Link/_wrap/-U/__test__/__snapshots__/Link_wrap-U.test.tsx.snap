// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link_wrap-U should render a content 1`] = `"<u class=\\"Link-U Link_wrap-U\\">Link with more clickable zone</u>"`;

exports[`Link_wrap-U should render a content 2`] = `"<u class=\\"Link-U Link_wrap-U Link_pseudo-U Link_pseudoBlack-U\\">Link with more clickable zone</u>"`;
